arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/CallSection.java

```diff
@@ -623,15 +692,56 @@ private void emptyCodeFirstCoupleOfAccountFragments(final Hub hub) {
     this.addFragments(firstCallerAccountFragment, firstCalleeAccountFragment);
   }
 
-  private MemorySpan returnDataMemorySpan(MessageFrame currentFrame, boolean callCanTransferValue) {
-    final int returnDataOffset =
-        callCanTransferValue
-            ? bytesToInt(currentFrame.getStackItem(5))
-            : bytesToInt(currentFrame.getStackItem(4));
-    final int returnDataLength =
-        callCanTransferValue
-            ? bytesToInt(currentFrame.getStackItem(6))
-            : bytesToInt(currentFrame.getStackItem(5));
-    return MemorySpan.fromStartLength(returnDataOffset, returnDataLength);
+  private MemorySpan callDataSpan(MessageFrame frame, boolean callHasValueArgument) {
+    final long callDataSize =
+        callHasValueArgument
+            ? Words.clampedToLong(frame.getStackItem(4))
+            : Words.clampedToLong(frame.getStackItem(3));
+
+    if (callDataSize == 0) {
+      return MemorySpan.empty();
+    }
```

Comment:
The "zero size, huge offset business.

Reply:
Note: `clampedToLong` is non destructive for 8 Byte integers or so. The MXPX exception, where we really need the actual values, kicks in at 4 or 6 byte sizes.

Reply:
yes, but we must NOT use `clampedToInt()` but to long is ok, I do agree.

Reply:
I agree. Do we `clampToInt` somewhere by mistake @letypequividelespoubelles ?

Reply:
I don't think so